### PR TITLE
Replace some try-finally statements

### PR DESCRIPTION
**Pull request by cbuescher**
This change replaces some existing try-finally statements that close resources
in their finally block with the slightly more readable and safer try-with-resources.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/engine/Engine.java [hunk 1]
```diff
@@ -1424,10 +1424,6 @@ public DocIdAndVersion docIdAndVersion() {
 
         @Override
         public void close() {
-            release();
-        }
-
-        public void release() {
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/index/engine/Engine.java]: Replaced all calls to release() with close().

File: server/src/main/java/org/elasticsearch/index/analysis/Analysis.java [hunk 2]
```diff
@@ -247,15 +247,9 @@ public static CharArraySet getWordSet(Environment env, Settings settings, String
         }
     }
 
-    public static List<String> loadWordList(Reader reader, String comment) throws IOException {
+    private static List<String> loadWordList(Path path, String comment) throws IOException {
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/index/analysis/Analysis.java]: I don't think this method should be public. If not, we can create the reader inside.

**Comment by hub-cap** [comment]
A few weeks ago the @elastic/es-core-infra team discussed the use of try-with-resources, and due to some really funky edge cases (exceptions thrown in finally's  and some other odd conditions in OOM) we decided that the benefits of code readability and inferred safety did not outweigh the possible problems associated with them. I will not +1 or -1 this, I am just bringing it up for posterity.

**Review by hub-cap — APPROVED** [review]
disregard, I mis-re-read the email. I <3 u @cbuescher and didnt mean to cause confusion!

**Comment by cbuescher** [comment]
@hub-cap thanks for pointing that out, I just checked the discussion out of curiosity, very informative.

**Pull request merged**